### DISCUSSION TITLE
BTT SKR 3 - ESP32U Firmware upload not working

### DISCUSSION THREAD

**Discussion by SpankyDevil** [luc-github/ESP3D#1032] [General]
I have downloaded ESP3D-2.1.4 and compiled

As per BTT SKR 3 instructions I have enabled the Serial Port in Marlin and renamed firmware.bin from the esp32dev folder and placed it on the root of my SD Card as esp3d.bin

Turned off the power and turned back on with the SD Card inserted, the status light flashes for a while but then when checking the SD Card the file hasn't been renamed to esp3d.cur

Checking there is a Wifi SSID ESP_E95E79 but this is an open Wifi does not require a password and when connected I can ping 192.168.4.1 but cannot open a webpage or Telnet

![image](https://github.com/luc-github/ESP3D/assets/75092373/49070e40-3870-4571-9cbf-efc889d9d185)
![image](https://github.com/luc-github/ESP3D/assets/75092373/6b6ce5ea-1ecf-401c-9615-7e2a2668ee7d)
![image](https://github.com/luc-github/ESP3D/assets/75092373/8f08467d-9516-4bba-b1af-24f5d41ad02f)
![image](https://github.com/luc-github/ESP3D/assets/75092373/96227958-b7ff-483a-b248-02b49fa92816)

**Comment by luc-github** [comment]
> the file hasn't been renamed to esp3d.cur

so the flashing failed , the fw you have is BTT one

**Reply by SpankyDevil** [reply]
> > the file hasn't been renamed to esp3d.cur
> 
> so the flashing failed , the fw you have is BTT one

Not sure i understand what you mean by the firmware i have is BTT one.

What do i need to do to resolve the issue?

**Reply by luc-github** [reply]
I just want to confirm the situation - you already have BTT version of esp3d Firmware on esp board and because update failed so you still have version of BTT of esp3d 
is that correct ?

**Reply by SpankyDevil** [reply]
> I just want to confirm the situation - you already have BTT version of esp3d Firmware on esp board and because update failed so you still have version of BTT of esp3d is that correct ?

I do not know if BTT have there own firmware version for this device as there instructions point to this github for the firmware but i can confirm this is BTT device and has not been updated or changed as of yet.

**Reply by SpankyDevil** [reply]
I also tried to update to version 3.0  and use the configurator tool but had the same issue

**Reply by luc-github** [reply]
if not updated it is original one fw provided by btt...
Anyway if update failed using BTT method you have to manually update / flash the esp board using usb/serial adapter

**Reply by SpankyDevil** [reply]
> if not updated it is original one fw provided by btt... Anyway if update failed using BTT method you have to manually update / flash the esp board

Do you have instructions for this method of update?

**Reply by luc-github** [reply]
everything is on http://esp3d.io  :
https://esp3d.io/esp3d/v2.x/installation/index.html
https://esp3d.io/esp3d/v3.x/installation/index.html


https://esp3d.io/esp3d/v3.x/hardware/esp_boards/esp32/btt-esp32e/index.html
https://esp3d.io/esp3d/v3.x/hardware/esp_boards/esp32/btt-esp32u/index.html

https://esp3d.io/esp3d/v3.x/installation/wiring/index.html

**Reply by SpankyDevil** [reply]
Sorry i dont follow, those links all describe connecting the ESP32 to the board and show pins etc.
That part has already been completed and is very straight forward with the SKR 3 as it just plugs in.

How do i flash the firmware.bin file if not using the BTT method through SD Card?

**Reply by SpankyDevil** [reply]
If my Config file is set incorrectly could that cause firmware to fail to update?
![Screenshot_20240630_203038_File Manager +](https://github.com/luc-github/ESP3D/assets/75092373/9315e916-f597-40aa-b783-f7793598f731)

**Reply by luc-github** [reply]
no this is related to original FW and BTT method - your  fw is not flashed yet so even incorrect it does not affect the  process of flashing

**Comment by luc-github** [comment]
you need serial usb adapter like the btt writer https://github.com/bigtreetech/BTT-WRITER-V1.0

**Reply by SpankyDevil** [reply]
Can i not update the firmware through the firmware that is currently installed which is broadcasting SSID ESP_E95E79 open Wifi does not require a password and ESP device IP 192.168.4.1

**Reply by luc-github** [reply]
You can try but I do not know if it will work on BTT FW, I do not know what they put by default

**Reply by SpankyDevil** [reply]
Not sure how i would upload to it as there is no web interface, telnet, ssh

**Reply by luc-github** [reply]
192.168.4.1 show what when you connect ?

**Reply by SpankyDevil** [reply]
> 192.168.4.1 show what when you connect ?

Connect with what application?
![image](https://github.com/luc-github/ESP3D/assets/75092373/b78df365-7469-429c-8f90-ed375f286c43)

![image](https://github.com/luc-github/ESP3D/assets/75092373/7b0d71be-d5b9-4b3e-a450-72ffd698420f)

![image](https://github.com/luc-github/ESP3D/assets/75092373/217a03d8-12c1-441f-8df7-2d0b66be2395)

**Reply by luc-github** [reply]
web browser , if no, then no choice flash manually

**Reply by SpankyDevil** [reply]
Any idea where I find the correct setting for my BTT ESP32U says on the device (ESP-wRoom-32U) 

![image](https://github.com/luc-github/ESP3D/assets/75092373/68b8633b-0a1c-475a-ad71-af6cd30be359)

**Reply by luc-github** [reply]
same as the ones I have set in platfomio.ini for esp32

**Reply by SpankyDevil** [reply]
Which pin is pin 1?

![image](https://github.com/luc-github/ESP3D/assets/75092373/75837d98-1b64-4b70-a283-16d625c27fd4)

**Reply by luc-github** [reply]
you have the pinout just right there - what you need more ?

**Reply by SpankyDevil** [reply]
From the picture of the board Is the top left pin RST?

also I'm struggling to work out these settings from file

![image](https://github.com/luc-github/ESP3D/assets/75092373/df9d537c-a843-469d-8564-c69912eb7e6a)

![image](https://github.com/luc-github/ESP3D/assets/75092373/a9ae4d36-c764-4507-9257-5ecc8e803f28)

**Comment by luc-github** [comment]
1 means row 1 and 2 means row 2 to match picture of the board 

>also I'm struggling to work out these settings from file

In that case why using this tool and not following the instructions I have linked to you ?

**Reply by SpankyDevil** [reply]
I am following the BTT Writer instructions you linked me,
It's just not the same ESP so settings are different.

**Reply by luc-github** [reply]
no I linked you the bttwriter to show you what kind of hardware you need to flash your ESP

**Reply by SpankyDevil** [reply]
Oh.
So just upload with Visual Studio Code once connected with USB to TTL device?

**Reply by luc-github** [reply]
yes as explained in documentation

**Reply by SpankyDevil** [reply]
> yes as explained in documentation

not that I saw, or could find. I've been hunting for those instructions all day.

**Reply by luc-github** [reply]
well link to documentation is in readme at least 5 or 6 times - can't do more sorry

**Reply by SpankyDevil** [reply]
well, that is very annoying i had looked over this at least 10 times today and couldn't find any info on installation only the connection.

and now I just realised you have to click the picture of the software :(

Well this should make things a lot easier

**Reply by SpankyDevil** [reply]
Thank You for your help.

**Reply by luc-github** [reply]
> well, that is very annoying i had looked over this at least 10 times today and couldn't find any info on installation only the connection.
> 
> and now I just realised you have to click the picture of the software :(
> 
> Well this should make things a lot easier

well not only ... there is a menu ...
![image](https://github.com/luc-github/ESP3D/assets/8822552/d9687f1b-7552-426d-bea2-d6cc17a81e30)  

and it is written to click on image ...
![image](https://github.com/luc-github/ESP3D/assets/8822552/935e66c1-8bef-44cc-8cfe-7e105f820711)

**Reply by SpankyDevil** [reply]
The wiring was wrong. its not as per the pictures I linked above RST is not the top left pin as per the photo :(

**Comment by SpankyDevil** [comment]
Still having issues, Cant connect to upload the firmware using VSCode
ESP32U Red Light is on Solid

Any advise?

![image](https://github.com/luc-github/ESP3D/assets/75092373/f898b8b5-2992-4ae8-a7f4-1a7fb79bf277)
![070e5cce0b59d32f539933d5fd02c3e1a47eb6e8](https://github.com/luc-github/ESP3D/assets/75092373/dc544caa-1c7e-4cf8-8d2f-80e5352b6fd5)
![image](https://github.com/luc-github/ESP3D/assets/75092373/a7b2d661-5b7d-4d18-9113-ed2ac872f07f)

**Comment by SpankyDevil** [comment]
Swapped Tx and Rx and managed to flash.
